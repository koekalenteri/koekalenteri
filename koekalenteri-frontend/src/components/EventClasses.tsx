import { CheckBox, CheckBoxOutlineBlank } from "@mui/icons-material";
import { Autocomplete, AutocompleteChangeReason, Avatar, Checkbox, Chip, TextField } from "@mui/material";
import { isSameDay } from "date-fns";
import { t } from "i18next";
import { EventClass } from "koekalenteri-shared/model";


/**
 * Callback fired when the value changes.
 *
 * @param {React.SyntheticEvent} event The event source of the callback.
 * @param {T|T[]} value The new value of the component.
 * @param {string} reason One of "createOption", "selectOption", "removeOption", "blur" or "clear".
 */
type EventClassesOnChange = (
  event: React.SyntheticEvent,
  value: EventClass[],
  reason: AutocompleteChangeReason
) => void;


type EventClassesProps = {
  id: string
  event: { classes?: EventClass[], startDate: Date }
  value: EventClass[] | undefined
  classes: EventClass[]
  label: string
  onChange: EventClassesOnChange
}

export const compareEventClass = (a: EventClass, b: EventClass) =>
  isSameDay(a.date || new Date(), b.date || new Date())
    ? a.class.localeCompare(b.class)
    : (a.date?.valueOf() || 0) - (b.date?.valueOf() || 0);

export function EventClasses({ id, event, value, classes, label, onChange }: EventClassesProps) {
  if (value) {
    value.sort(compareEventClass);
  }

  return (
    <Autocomplete
      id={id}
      fullWidth
      disableClearable
      disableCloseOnSelect
      disabled={classes.length === 0}
      multiple
      value={value}
      groupBy={c => t('weekday', { date: c.date })}
      options={classes}
      onChange={onChange}
      getOptionLabel={c => c.class}
      isOptionEqualToValue={(o, v) => compareEventClass(o, v) === 0}
      renderOption={(props, option, { selected }) => (
        <li {...props}>
          <Checkbox
            icon={<CheckBoxOutlineBlank fontSize="small" />}
            checkedIcon={<CheckBox fontSize="small" />}
            style={{ marginRight: 8 }}
            checked={selected}
          />
          {option.class}
        </li>
      )}
      renderInput={(params) => <TextField {...params} label={label} />}
      renderTags={(tagValue, getTagProps) => tagValue.map((option, index) => (
        <Chip
          {...getTagProps({ index })}
          avatar={
            <Avatar
              sx={{
                fontWeight: 'bold',
                bgcolor: isSameDay(option.date || event.startDate, event.startDate) ? 'secondary.light' : 'secondary.dark'
              }}
            >
              {t('weekday', { date: option.date })}
            </Avatar>
          }
          label={option.class}
          onDelete={undefined}
          size="small"
          sx={{bgcolor: option.judge?.id ? 'background.ok' : 'transparent'}}
          variant={option.judge ? "filled" : "outlined"}
        />
      ))}
    />
  );
}
