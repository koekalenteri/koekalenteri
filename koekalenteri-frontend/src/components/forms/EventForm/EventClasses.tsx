import { useTranslation } from "react-i18next"
import { CheckBox, CheckBoxOutlineBlank } from "@mui/icons-material"
import { Autocomplete, AutocompleteChangeReason, Avatar, Checkbox, Chip, TextField } from "@mui/material"
import { isSameDay } from "date-fns"
import { Event, EventClass, EventState } from "koekalenteri-shared/model"

import { PartialEvent } from "../.."

/**
 * Callback fired when the value changes.
 *
 * @param {React.SyntheticEvent} event The event source of the callback.
 * @param {T|T[]} value The new value of the component.
 * @param {string} reason One of "createOption", "selectOption", "removeOption", "blur" or "clear".
 */
type EventClassesOnChange = (
  event: React.SyntheticEvent,
  value: EventClass[],
  reason: AutocompleteChangeReason
) => void


type EventClassesProps = {
  id: string
  event: PartialEvent
  value: EventClass[] | undefined
  classes: EventClass[]
  label: string
  required?: boolean
  requiredState?: EventState
  errorStates?: { [Property in keyof Event]?: boolean }
  helperTexts?: { [Property in keyof Event]?: string }
  onChange: EventClassesOnChange
}

export const compareEventClass = (a: EventClass, b: EventClass) =>
  isSameDay(a.date || new Date(), b.date || new Date())
    ? a.class.localeCompare(b.class)
    : (a.date?.valueOf() || 0) - (b.date?.valueOf() || 0)

export function EventClasses(props: EventClassesProps) {
  const { t } = useTranslation()
  const { classes, label, event, required, requiredState, errorStates, helperTexts, value, ...rest } = props
  const error = errorStates?.classes
  const helperText = helperTexts?.classes || ''
  const sortedValue = value?.slice().sort(compareEventClass)

  return (
    <Autocomplete
      {...rest}
      value={sortedValue}
      fullWidth
      disableClearable
      disableCloseOnSelect
      disabled={classes.length === 0}
      multiple
      groupBy={c => t('weekday', { date: c.date })}
      options={classes}
      getOptionLabel={c => c.class}
      isOptionEqualToValue={(o, v) => compareEventClass(o, v) === 0}
      renderOption={(optionProps, option, { selected }) => (
        <li {...optionProps}>
          <Checkbox
            icon={<CheckBoxOutlineBlank fontSize="small" />}
            checkedIcon={<CheckBox fontSize="small" />}
            style={{ marginRight: 8 }}
            checked={selected}
          />
          {option.class}
        </li>
      )}
      renderInput={(inputProps) => <TextField {...inputProps} required={required} error={!!error} helperText={helperText} label={label} />}
      renderTags={(tagValue, getTagProps) => tagValue.map((option, index) => (
        <Chip
          {...getTagProps({ index })}
          avatar={
            <Avatar
              sx={{
                fontWeight: 'bold',
                bgcolor: isSameDay(option.date || event.startDate, event.startDate) ? 'secondary.light' : 'secondary.dark',
              }}
            >
              {t('weekday', { date: option.date })}
            </Avatar>
          }
          label={option.class}
          onDelete={undefined}
          size="small"
          sx={{bgcolor: option.judge?.id ? 'background.ok' : 'transparent'}}
          variant={option.judge ? "filled" : "outlined"}
        />
      ))}
    />
  )
}
