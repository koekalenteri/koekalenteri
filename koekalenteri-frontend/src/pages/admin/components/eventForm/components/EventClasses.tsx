import { SyntheticEvent, useCallback, useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import { CheckBox, CheckBoxOutlineBlank } from '@mui/icons-material'
import { Autocomplete, AutocompleteChangeReason, Avatar, Checkbox, Chip, TextField } from '@mui/material'
import { isSameDay } from 'date-fns'
import { DeepPartial, Event, EventClass, EventState } from 'koekalenteri-shared/model'

interface Props {
  id: string
  eventStartDate: Date
  eventEndDate: Date
  value: DeepPartial<EventClass>[] | undefined
  classes: DeepPartial<EventClass>[]
  label: string
  required?: boolean
  requiredState?: EventState
  errorStates?: { [Property in keyof Event]?: boolean }
  helperTexts?: { [Property in keyof Event]?: string }
  showCount?: boolean
  onChange?: (event: SyntheticEvent, value: DeepPartial<EventClass>[], reason: AutocompleteChangeReason) => void
}

export const compareEventClass = (a: DeepPartial<EventClass>, b: DeepPartial<EventClass>) =>
  isSameDay(a.date ?? new Date(), b.date ?? new Date())
    ? a.class?.localeCompare(b.class ?? '') ?? 0
    : (a.date?.valueOf() ?? 0) - (b.date?.valueOf() ?? 0)

export default function EventClasses(props: Props) {
  const { t } = useTranslation()
  const {
    classes,
    label,
    eventStartDate,
    eventEndDate,
    required,
    requiredState,
    errorStates,
    helperTexts,
    value,
    showCount,
    ...rest
  } = props
  const error = errorStates?.classes
  const helperText = helperTexts?.classes ?? ''
  const sortedValue = useMemo(() => value?.slice().sort(compareEventClass), [value])
  const groupByWeekday = useCallback((c: { date?: Date }) => t('dateFormat.wdshort', { date: c.date }), [t])
  const getLabel = useCallback((c: { class?: string }) => c.class ?? '', [])
  const isEqual = useCallback(
    (a: DeepPartial<EventClass>, b: DeepPartial<EventClass>) => compareEventClass(a, b) === 0,
    []
  )
  const groupBy = !isSameDay(eventStartDate, eventEndDate)
  const hasJudge = useCallback(
    (option: DeepPartial<EventClass>) => (Array.isArray(option.judge) ? option.judge.length : option.judge?.id),
    []
  )

  return (
    <Autocomplete
      {...rest}
      value={sortedValue}
      fullWidth
      disableClearable
      disableCloseOnSelect
      disabled={classes.length === 0}
      disablePortal
      multiple
      groupBy={groupBy ? groupByWeekday : undefined}
      options={classes}
      getOptionLabel={getLabel}
      isOptionEqualToValue={isEqual}
      renderOption={(optionProps, option, { selected }) => (
        <li {...optionProps}>
          <Checkbox
            icon={<CheckBoxOutlineBlank fontSize="small" />}
            checkedIcon={<CheckBox fontSize="small" />}
            style={{ marginRight: 8 }}
            checked={selected}
          />
          {option.class}
        </li>
      )}
      renderInput={(inputProps) => (
        <TextField {...inputProps} required={required} error={!!error} helperText={helperText} label={label} />
      )}
      renderTags={(tagValue, getTagProps) =>
        tagValue.map((option, index) => (
          <Chip
            {...getTagProps({ index })}
            avatar={
              groupBy ? (
                <Avatar
                  sx={{
                    fontWeight: 'bold',
                    bgcolor: isSameDay(option.date || eventStartDate, eventStartDate)
                      ? 'secondary.light'
                      : 'secondary.dark',
                  }}
                >
                  {t('dateFormat.weekday', { date: option.date })}
                </Avatar>
              ) : undefined
            }
            label={
              (option.class ?? '') +
              (showCount && Array.isArray(option.judge) && option.judge.length > 1 ? ` x${option.judge.length}` : '')
            }
            onDelete={undefined}
            size="small"
            sx={{ bgcolor: hasJudge(option) ? 'background.ok' : 'transparent' }}
            variant={hasJudge(option) ? 'filled' : 'outlined'}
          />
        ))
      }
    />
  )
}
