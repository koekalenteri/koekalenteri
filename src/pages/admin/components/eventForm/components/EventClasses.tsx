import type { AutocompleteChangeReason } from '@mui/material'
import type { SyntheticEvent } from 'react'
import type { DeepPartial, DogEvent, EventClass, EventState } from '../../../../../types'

import { useCallback, useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import CheckBox from '@mui/icons-material/CheckBox'
import CheckBoxOutlineBlank from '@mui/icons-material/CheckBoxOutlineBlank'
import Autocomplete from '@mui/material/Autocomplete'
import Avatar from '@mui/material/Avatar'
import Checkbox from '@mui/material/Checkbox'
import Chip from '@mui/material/Chip'
import TextField from '@mui/material/TextField'
import { isSameDay } from 'date-fns'

interface Props {
  readonly id: string
  readonly eventStartDate: Date
  readonly eventEndDate: Date
  readonly value: DeepPartial<EventClass>[] | undefined
  readonly classes: DeepPartial<EventClass>[]
  readonly label: string
  readonly disabled?: boolean
  readonly required?: boolean
  readonly requiredState?: EventState
  readonly errorStates?: { [Property in keyof DogEvent]?: boolean }
  readonly helperTexts?: { [Property in keyof DogEvent]?: string }
  readonly showCount?: boolean
  readonly onChange?: (
    event: SyntheticEvent,
    value: readonly DeepPartial<EventClass>[],
    reason: AutocompleteChangeReason
  ) => void
}

export const compareEventClass = (a: DeepPartial<EventClass>, b: DeepPartial<EventClass>) =>
  isSameDay(a.date ?? new Date(), b.date ?? new Date())
    ? (a.class?.localeCompare(b.class ?? '') ?? 0)
    : (a.date?.valueOf() ?? 0) - (b.date?.valueOf() ?? 0)

export default function EventClasses(props: Props) {
  const { t } = useTranslation()
  const {
    classes,
    label,
    eventStartDate,
    eventEndDate,
    disabled,
    required,
    requiredState,
    errorStates,
    helperTexts,
    value,
    showCount,
    ...rest
  } = props
  const error = errorStates?.classes
  const helperText = helperTexts?.classes ?? ''
  const sortedValue = useMemo(() => value?.slice().sort(compareEventClass), [value])
  const groupByWeekday = useCallback((c: { date?: Date }) => t('dateFormat.wdshort', { date: c.date }), [t])
  const getLabel = useCallback((c: { class?: string }) => c.class ?? '', [])
  const isEqual = useCallback(
    (a: DeepPartial<EventClass>, b: DeepPartial<EventClass>) => compareEventClass(a, b) === 0,
    []
  )
  const groupBy = !isSameDay(eventStartDate, eventEndDate)
  const hasJudge = useCallback(
    (option: DeepPartial<EventClass>) => (Array.isArray(option.judge) ? option.judge.length : option.judge?.id),
    []
  )

  return (
    <Autocomplete
      {...rest}
      value={sortedValue}
      fullWidth
      disableClearable
      disableCloseOnSelect
      disabled={disabled || classes.length === 0}
      disablePortal
      multiple
      groupBy={groupBy ? groupByWeekday : undefined}
      options={classes}
      getOptionLabel={getLabel}
      isOptionEqualToValue={isEqual}
      renderOption={(optionProps, option, { selected }) => (
        <li {...optionProps} key={`${option.date}-${option.class}`}>
          <Checkbox
            icon={<CheckBoxOutlineBlank fontSize="small" />}
            checkedIcon={<CheckBox fontSize="small" />}
            style={{ marginRight: 8 }}
            checked={selected}
          />
          {option.class}
        </li>
      )}
      renderInput={(inputProps) => (
        <TextField {...inputProps} required={required} error={!!error} helperText={helperText} label={label} />
      )}
      renderTags={(tagValue, getTagProps) =>
        tagValue.map((option, index) => {
          const { key, ...props } = getTagProps({ index })
          return (
            <Chip
              key={key}
              {...props}
              avatar={
                groupBy ? (
                  <Avatar
                    sx={{
                      fontWeight: 'bold',
                      bgcolor: isSameDay(option.date || eventStartDate, eventStartDate)
                        ? 'secondary.light'
                        : 'secondary.dark',
                    }}
                  >
                    {t('dateFormat.weekday', { date: option.date })}
                  </Avatar>
                ) : undefined
              }
              label={
                (option.class ?? '') +
                (showCount && Array.isArray(option.judge) && option.judge.length > 1 ? ` x${option.judge.length}` : '')
              }
              onDelete={undefined}
              size="small"
              sx={{ bgcolor: hasJudge(option) ? 'background.ok' : 'transparent' }}
              variant={hasJudge(option) ? 'filled' : 'outlined'}
            />
          )
        })
      }
    />
  )
}
