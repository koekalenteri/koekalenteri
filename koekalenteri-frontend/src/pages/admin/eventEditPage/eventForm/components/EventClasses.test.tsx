
import { render } from '@testing-library/react'
import { parseISO } from 'date-fns'

import EventClasses from './EventClasses'

const date = parseISO('2023-01-17')

describe('EventClasses', () => {
  it('should render with minimal properties', () => {
    const { container } = render(<EventClasses id={''} eventStartDate={date} value={undefined} classes={[]} label={''} />)
    expect(container).toMatchSnapshot()
  })

  it('should render with classes', () => {
    const { container } = render(<EventClasses
      id={''}
      eventStartDate={date}
      value={undefined}
      classes={[{class: 'ALO'}, {class: 'AVO'}]}
      label={''}
    />)
    expect(container).toMatchSnapshot()
  })

  it('should render with classes and value', () => {
    const { container } = render(<EventClasses
      id={''}
      eventStartDate={date}
      value={[{class: 'ALO', date, judge: {id: 1, name: 'Test Judge'}}]}
      classes={[{class: 'ALO', date}, {class: 'AVO', date}]}
      label={''} />)
    expect(container).toMatchSnapshot()
  })

  it('should render with classes and values', () => {
    const { container } = render(<EventClasses
      id={''}
      eventStartDate={date}
      value={[
        {class: 'ALO', date, judge: {id: 1, name: 'Test Judge'}},
        {class: 'AVO', date, judge: [{id: 1, name: 'Test Judge'}, {id: 2, name: 'Test Judge2'}]},
      ]}
      classes={[{class: 'ALO', date}, {class: 'AVO', date}]}
      label={''}
      showCount
    />)
    expect(container).toMatchSnapshot()
  })
})
