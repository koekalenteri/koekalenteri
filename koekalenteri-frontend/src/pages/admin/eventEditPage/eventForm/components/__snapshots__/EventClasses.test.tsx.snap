// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventClasses should render with classes 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-qm9mjt-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventClasses should render with classes and value 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-qm9mjt-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-1u2nxr-MuiChip-root"
          data-tag-index="0"
          tabindex="-1"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault MuiChip-avatar MuiChip-avatarSmall MuiChip-avatarColorDefault css-y3b1cu-MuiAvatar-root"
          >
            weekday
          </div>
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            ALO
          </span>
        </div>
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventClasses should render with classes and values 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-qm9mjt-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-1u2nxr-MuiChip-root"
          data-tag-index="0"
          tabindex="-1"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault MuiChip-avatar MuiChip-avatarSmall MuiChip-avatarColorDefault css-y3b1cu-MuiAvatar-root"
          >
            weekday
          </div>
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            ALO
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-1u2nxr-MuiChip-root"
          data-tag-index="1"
          tabindex="-1"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault MuiChip-avatar MuiChip-avatarSmall MuiChip-avatarColorDefault css-y3b1cu-MuiAvatar-root"
          >
            weekday
          </div>
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            AVO x2
          </span>
        </div>
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventClasses should render with minimal properties 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-qm9mjt-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          disabled=""
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            disabled=""
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
